vscode/src/jdkDownloader/view.ts

```diff
@@ -234,6 +238,28 @@ export class JdkDownloaderView {
                     triggerJDKDownload(event);
                 });
 
+                document.getElementById("oracleJDKOsTypeDropdown")?.addEventListener('change', (event) => {
+                    updateMachineArchOptions(event.target.id);
+                });
+
+                document.getElementById("openJDKOsTypeDropdown")?.addEventListener('change', (event) => {
+                    updateMachineArchOptions(event.target.id);
+                });
+                
+                const updateMachineArchOptions = (osDropdownId) => {
+                    const osDropdown = document.getElementById(osDropdownId);
+                    const machineArchDropdown = document.getElementById(osDropdownId.replace('OsTypeDropdown', 'MachineArchDropdown'));
+                    
+                    if (osDropdown.value === 'windows') {
+                        machineArchDropdown.innerHTML = '<option value="x64" selected>x64</option>';
+                    } else {
+                        machineArchDropdown.innerHTML = \`
+                            <option value="aarch64" \${machineArchDropdown.value === 'aarch64' ? 'selected' : null}>arm64</option>
+                            <option value="x64" \${machineArchDropdown.value === 'x64' ? 'selected' : null}>x64</option>
+                        \`;
+                    }
```

Comment:
I didn't understand why `getMachineArchHtml()` could not be used here directly.
Some doubts:
1. If a user switches from windows to linux/mac, 
    - Should the dropdown continue to use the previously selected dropdown value? 
    - Its possible that the machine's arch is ARM and so that should be given preference.
2. If a user switches to windows, 
    - then the `getMachineArchHtml` function would handle this appropriately.
3. If a user switches between linux and mac, 
    - then the possibility of the selected OS not being the host's OS is present for both the source and the target OS. 
    - If target OS is the host OS, then we should prefer selecting the host arch. 
    - Else, it is possible that the user wants to maintain the selected arch value. 
    - Should we enhance `getMachineArchHtml` to handle this?
    - IMO perhaps we can just keep it simple and just use the host's arch to make the selection.

Reply:
> * Should the dropdown continue to use the previously selected dropdown value?

IMO user would have chosen something then we should honour that and shouldn't change according to the machine architecture. By default, it is machine architecture. But, if you are not convinced I can change the behaviour no issues.




> * IMO perhaps we can just keep it simple and just use the host's arch to make the selection.

IMO we can at the start keep defaults, and after that let user only do the selection and change part and don't change according to us, it will keep the things simple also.